contentcuration/contentcuration/views.py

```diff
@@ -275,55 +277,98 @@ def duplicate_nodes(request):
+
+    # create nodes, one level at a time, starting from the top of the tree (so that we have IDs to pass as "parent" for next level down)
+    for node_level in to_create["nodes"]:
+        for node in node_level:
+            node.parent_id = node.parent.id
+        ContentNode.objects.bulk_create(node_level)
+        for node in node_level:
+            for tag in node._meta.tags_to_add:
+                node.tags.add(tag)
+
+    # rebuild MPTT tree for this channel (since we're inside "disable_mptt_updates", and bulk_create doesn't trigger rebuild signals anyway)
+    ContentNode.objects.partial_rebuild(to_create["nodes"][0][0].tree_id)
+
+    # create each of the assessment items
+    for a in to_create["assessments"]:
+        a.contentnode_id = a.contentnode.id
+    AssessmentItem.objects.bulk_create(to_create["assessments"])
+
+    # create the file objects, for both nodes and assessment items
+    for f in to_create["node_files"]:
+        f.contentnode_id = f.contentnode.id
+    for f in to_create["assessment_files"]:
+        f.assessment_item_id = f.assessment_item.id
+    File.objects.bulk_create(to_create["node_files"] + to_create["assessment_files"])
+
+    return new_node
+
+def _duplicate_node_bulk_recursive(node, sort_order, parent, channel_id, to_create, level=0):
+
     if isinstance(node, int) or isinstance(node, basestring):
         node = ContentNode.objects.get(pk=node)
 
-    original_channel = node.get_original_node().get_channel() if node.get_original_node() else None
-
-    new_node = ContentNode.objects.create(
-        title=node.title,
-        description=node.description,
-        kind=node.kind,
-        license=node.license,
-        parent=ContentNode.objects.get(pk=parent) if parent else None,
-        sort_order=sort_order or node.sort_order,
-        copyright_holder=node.copyright_holder,
-        changed=True,
-        original_node=node.original_node or node,
-        cloned_source=node,
-        original_channel_id = node.original_channel_id or original_channel.id if original_channel else None,
-        source_channel_id = node.get_channel().id if node.get_channel() else None,
-        original_source_node_id = node.original_source_node_id or node.node_id,
-        source_node_id = node.node_id,
-        author=node.author,
-        content_id=node.content_id,
-        extra_fields=node.extra_fields,
-    )
-
-    # add tags now
+    if isinstance(parent, int) or isinstance(parent, basestring):
+        parent = ContentNode.objects.get(pk=parent)
+
+    # clone the model (in-memory) and update the fields on the cloned model
+    new_node = copy.copy(node)
+    new_node.id = None
+    new_node.tree_id = parent.tree_id
+    new_node.parent = parent
+    new_node.sort_order = sort_order or node.sort_order
+    new_node.changed = True
+    new_node.cloned_source = node
+    new_node.source_channel_id = channel_id
```

Comment:
@jayoshih Could you fix this to use a `source_channel_id` value (right now uses target channel's ID)